RecoLocalTracker/SiStripClusterizer/plugins/alpaka/SiStripClusterizerConditionsESProducerAlpaka.cc

```diff
@@ -0,0 +1,225 @@
+
+      // Prepare the conditions on the host
+      auto product = std::make_unique<SiStripClusterizerConditionsDataHostObject>(cms::alpakatools::host());
+      const int Data_fedch_size = (*product)->invthick.size();  // 42240
+      const int Data_strip_size = (*product)->noise.size();     // 10813440
+      const int Data_apv_size = (*product)->gain.size();        // 84480
+
+      // Fill the collections
+      fillSiStripClusterizerConditions(quality,
+                                       gains.product(),
+                                       noises,
+                                       std::span((*product)->invthick.data(), Data_fedch_size),
+                                       std::span((*product)->detID.data(), Data_fedch_size),
+                                       std::span((*product)->iPair.data(), Data_fedch_size),
+                                       std::span((*product)->noise.data(), Data_strip_size),
+                                       std::span((*product)->gain.data(), Data_apv_size));
+      //
+      return product;
+    }
+
+    // Auxiliary functions to translate indexes on the arrays
+    static constexpr uint16_t badBit = 1 << 15;
+    inline uint16_t fedIndex(uint16_t fed) { return (fed - FED_ID_MIN); };
+    inline uint32_t stripIndex(uint16_t fed, uint16_t channel, uint16_t strip) {
+      return (fedIndex(fed) * FEDCH_PER_FED * STRIPS_PER_FEDCH + channel * STRIPS_PER_FEDCH +
+              (strip % STRIPS_PER_FEDCH));
+    };
+    inline uint32_t apvIndex(uint16_t fed, uint16_t channel, uint16_t strip) {
+      return (fedIndex(fed) * APVS_PER_FEDCH * FEDCH_PER_FED + APVS_PER_CHAN * channel +
+              (strip % STRIPS_PER_FEDCH) / STRIPS_PER_APV);
+    };
+    inline uint32_t channelIndex(uint16_t fedId, uint16_t fedCh) { return (fedIndex(fedId) * FEDCH_PER_FED + fedCh); };
+
+  private:
+    edm::ESGetToken<SiStripGain, SiStripGainRcd> gainsToken_;
+    edm::ESGetToken<SiStripNoises, SiStripNoisesRcd> noisesToken_;
+    // Two tokens for quality because of the exc message:
+    // You may need multiple tokens if you want to get the same data in multiple transitions.
+    edm::ESGetToken<SiStripQuality, SiStripQualityRcd> qualityTokenA_;
+    edm::ESGetToken<SiStripQuality, SiStripQualityRcd> qualityTokenB_;
+
+    // Make conditions as in the RecoLocalTracker/SiStripClusterizer/plugins/ClustersFromRawProducer.cc module
+    void fillSiStripClusterizerConditions(const SiStripQuality& quality,
+                                          const SiStripGain* gains,
+                                          const SiStripNoises& noises,
+                                          std::span<float> invthick,
+                                          std::span<uint32_t> detID,
+                                          std::span<uint16_t> iPair,
+                                          std::span<uint16_t> noise,
+                                          std::span<float> gain);
```

Comment:
I think you could simplify a bit the code by passing these as `std::array<...> &` ?
Or even as a whole `sistrip::Data &` ?